root/usr/bin/discovery-script

```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+basescript=${KCL_FDI_SCRIPT:-H4sIAOj1ZloCA0tNzshXUPLLVyhOLsosKFEoSCwuTk1R4gIAIbwTwhgAAAA=}
+echo "$basescript" | base64 -d | gunzip > /tmp/discovery-script
```

Comment:
I think there should be a little more error checking here.  COMMAND_LINE_SIZE isn't very big: 2K on my Fedora x86 box.  A script end up truncated, and I know it's an extreme and contrived example but "rm -rf /tmp/foo" could end up as "rm -rf /":

```
echo "rm -rf /tmp/foo" | gzip -9 | base64 -w0 | cut -c 1-26 | base64 -d | gunzip
```

Which I know isn't a big deal on the discovery image, but base64 will return an error code if the input isn't quite valid, can we check it?

Reply:
Ok it looks like if a single character is missing (except `=` which is safe to cut) gunzip return non-zero code. I will take advantage and fail.